dot/node.go

```diff
@@ -146,29 +146,27 @@ func NodeInitialized(basepath string) bool {
 }
 
 // LoadGlobalNodeName returns the stored global node name from database
-func LoadGlobalNodeName(basepath string) (string, error) {
+func LoadGlobalNodeName(basepath string) (nodename string, err error) {
 	// initialise database using data directory
 	db, err := utils.SetupDatabase(basepath, false)
 	if err != nil {
 		return "", err
 	}
 
 	defer func() {
-		err = db.Close()
-		if err != nil {
-			logger.Errorf("failed to close database: %s", err)
+		e := db.Close()
+		if e != nil {
+			logger.Errorf("failed to close database: %s", e)
 			return
 		}
 	}()
 
 	basestate := state.NewBaseState(db)
-	nodename, err := basestate.LoadNodeGlobalName()
+	nodename, err = basestate.LoadNodeGlobalName()
 	if err != nil {
 		logger.Warnf("failed to load global node name from base path %s: %s", basepath, err)
-		return "", err
 	}
-
-	return nodename, err
+	return
```

Comment:
No naked returns, it's really harder to read and leads to confusion

https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_naked_returns

To be fair, I have no clue why this is even part of the language...

Reply:
updated.